test/parser/nan-inf.jl

```diff
@@ -0,0 +1,14 @@
+@testset begin
+    test_str = """{"x": NaN, "y": Infinity, "z": -Infinity}"""
+    @test_throws ErrorException JSON.parse(test_str)
+    val = JSON.parse(test_str, allownan=true)
+    @test val["x"] === NaN
```

Comment:
I'm a little bugged about the === vs == here (also, not all NaNs are ===). Can we use `isnan` or `isequal` consistently?

```
julia> reinterpret(Float64, 9221120237041090561)
NaN

julia> reinterpret(Float64, 9221120237041090561) === NaN
false
```

Reply:
> I'm a little bugged about the `===` vs `==` here (also, not all `NaN`s are `===`). Can we use `isnan` or `isequal` consistently?

Sure.  I had forgotten that `isequal` compared treated `NaN`s as equal.

Reply:
Using `isequal` for all tests in the latest commit.